.github/workflows/coverage.yml

```diff
@@ -0,0 +1,46 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Huawei Inc.
+#
+name: Generate Coverage Badge
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set Up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: 17
+          distribution: adopt
+
+      - name: Test
+        run: mvn --batch-mode --update-snapshots --no-transfer-progress test
+
+      - name: Extract Coverage Percentage
+        run: |
+          coverage=$(grep -oP '<td class="ctr2">\K(\d+)%' target/coverage-reports/index.html | head -n 1)
+          coverage=${coverage%\%}
+          echo "COVERAGE_PERCENTAGE=$coverage" >> $GITHUB_ENV
+
+      - name: Create Coverage Badge
+        uses: schneegans/dynamic-badges-action@v1.6.0
+        with:
+          auth: ${{ secrets.GIST_TOKEN }}
+          gistID: 3d9c022b98734fbf615c21136abe4add
+          filename: gistfile1.txt
```

Comment:
need better name. `coverage.txt` ?

Reply:
![name](https://github.com/eclipse-xpanse/xpanse/assets/121535229/23c0da88-a6e3-41af-8335-5bd944aa1c56)

 coverage.txt is fine,little effect here